### DISCUSSION TITLE
固件是好固件，但是太精简了，网络连不上，想传文件上去，发现ssh都没文件管理

### DISCUSSION THREAD

**Discussion by Sevenzhi** [immortalwrt/immortalwrt#1520] [General]
### Describe the bug

能不能多增加一点功能啊，必要的自带的最好了，因为刚开始没网络，啥东西都安装不上啊，只能本地安装，却发现传文件都不行

### ImmortalWrt version

. /etc/openwrt_release && echo $DISTRIB_REVISION

### ImmortalWrt release

. /etc/openwrt_release && echo $DISTRIB_RELEASE

### ImmortalWrt target/subtarget

. /etc/openwrt_release && echo $DISTRIB_TARGET

### Device

cat /tmp/sysinfo/model

### Image kind

Official downloaded image

### Terms

- [X] I am reporting an issue for ImmortalWrt, not an unsupported fork.

**Comment by science2468** [comment]
去百度: winscp

**Reply by Sevenzhi** [reply]
我用的WindTerm shh连接后不能进行文件传输，提示缺少组件，我用其他的没这个问题

**Comment by libriunk** [comment]
你是否需要immo selector

**Reply by Sevenzhi** [reply]
没有升级功能

**Reply by libriunk** [reply]
> 没有升级功能

luci-app-sysupgrade还是什么，你可以搜一下

**Comment by science2468** [comment]
ssh里面有scp命令可以上传/下载文件，具体命令可以百度。
发送自 Windows 10 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>应用

发件人: ***@***.***>
发送时间: 2024年10月17日 22:02
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
主题: Re: [immortalwrt/immortalwrt] 固件是好固件，但是太精简了，网络连不上，想传文件上去，发现ssh都没文件管理 (Discussion #1520)


我用的WindTerm shh连接后不能进行文件传输，提示缺少组件，我用其他的没这个问题

―

**Comment by unipoco** [comment]
看下软件包里有没有openssh-sftp-server，缺少这个sftp会连不上

**Reply by bluceli0** [reply]
有效，过了这么长时间官方也不默认加上这个

**Reply by libriunk** [reply]
> 有效，过了这么长时间官方也不默认加上这个

ssh相关默认用的是dropbear

**Reply by docker-master** [reply]
牛逼，我用的final shell ，安装这个软件包就能看到文件了，这么小的软件包官方也不装一下

**Comment by theosoft-git** [comment]
scp要加-O参数

**Reply by lanten** [reply]
卧槽，管用！

**Comment by Profitter** [comment]
绝了软件包要fq才能更新，但我就是为了解决nikki插件缺少mmdb需要本地上传结果发现sftp也要fq才能安装，闭环了

**Reply by libriunk** [reply]
> 绝了软件包要fq才能更新，但我就是为了解决nikki插件缺少mmdb需要本地上传结果发现sftp也要fq才能安装，闭环了

有一个邪门办法，使用固件升级功能，上传的文件在`/tmp/firmware.bin`，ssh进去根据需求改成原文件名操作即可

**Reply by minortex** [reply]
自己换源啊[这个](https://help.mirrors.cernet.edu.cn/immortalwrt/)

**Comment by wukongdaily** [comment]
因为默认情况下没有集成 `openssh-sftp-server`
不过没网就不建议安装软件了。很多软件并非独立的，都需要一些依赖。而`opkg update` 操作必然需要网络
有网之后 安装什么软件都简单了

**Comment by victorious210** [comment]
需要sftp看这里https://openwrt.org/docs/guide-user/services/ssh/openssh_instead_dropbear
